### PR TITLE
Azure Login redesign

### PR DESCRIPTION
**Pull request by Tanya0609**
This is to address part of the requirements under #1111 

**How to test:**
1. start the extension
2. in Azure page
- [ ] Login section is removed 
- [ ] click Sign in button, it opens a sign in modal
- [ ] after sign in, the modal closes automatically and redirects back to Azure page  

Expected:
sign-in
sign-in-modal
sign-out

### CODE REVIEW

File: src/client/src/containers/AzureLoginModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+      closeModal();
+    }
+  };
+
+  React.useEffect(() => {
+    // close sign in modal when user logs in
+    if (isLoggedIn) {
+      closeModal();
+    }
+  }, [isLoggedIn]);
```
Comment (jsondoo) [inline src/client/src/containers/AzureLoginModal/index.tsx:60]: beautiful

Comment (Tanya0609) [reply src/client/src/containers/AzureLoginModal/index.tsx:60]: You helped me on this one haha 👍

**Comment by jasmingoh** [comment]
Cursor should be a pointer when hovering over the Log In/Create an Account 
image

**Comment by jasmingoh** [comment]
hmm TODO? 
image

File: src/client/src/containers/AzureLoginModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        <Cancel
+          tabIndex={0}
+          className={styles.cancelIcon}
+          onClick={props.closeModal}
+          onKeyDown={cancelKeyDownHandler}
+        />
+      </div>
+
+      <div className={styles.questionaryContainer}>
+        <CollapsibleInfoBox
```
Comment (SahilTara) [inline src/client/src/containers/AzureLoginModal/index.tsx:77]: Are these togglable via tab + enter? Just wondering.

File: src/client/src/containers/AzureLoginModal/strings.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { defineMessages } from "react-intl";
+
+export const strings = defineMessages({
+    freeAccountAnswer: {
+        defaultMessage: "All you need is a phone number, a credit or debit card, and a GitHub account or Microsoft account username (formerly Windows Live ID).",
+        id: "auzreLoginModal.freeAccountAnswer"
```
Comment (SahilTara) [inline src/client/src/containers/AzureLoginModal/strings.ts]: Nit: isn't it azureLoginModal, happens throughout the file.

**Comment by Tanya0609** [comment]
> hmm TODO?
> image

Was waiting for @xtinah-w to give me answer LOL good catch! PR updated

File: src/client/src/containers/AzureLoginModal/strings.ts (hunk 3 above)
Comment (Tanya0609) [reply src/client/src/containers/AzureLoginModal/strings.ts]: Good Catch! Thanks :D

**Comment by Tanya0609** [comment]
> Cursor should be a pointer when hovering over the Log In/Create an Account
> image

PR updated :)

File: src/client/src/containers/AzureLoginModal/index.tsx (hunk 2 above)
Comment (Tanya0609) [reply src/client/src/containers/AzureLoginModal/index.tsx:77]: Not at the moment

**Comment by xtinah-w** [comment]
What happens if I want to upgrade my free trial?
After 12 months, any of these products you may be using will continue to run, and you’ll be billed at the standard pay-as-you-go rates.

File: src/client/src/translations/en.json [hunk 4]
```diff
@@ -39,6 +39,18 @@
...
   "azureLogin.azureTitle": "Microsoft Azure",
   "azureLogin.longDescription": "Azure is a cloud computing service created by Microsoft for building, testing, deploying, and managing applications and services through Microsoft-managed data centers spread throughout the world. It offers more than a hundred services including scalable databases, container deployments, infrastructure management, serverless compute etc. These services combined with Microsoft's developer tools like Visual Studio and Visual Studio Code offer you great end-to-end tools to make you success. More information about Microsoft Azure can be found at [azure.microsoft.com](azure.microsoft.com).",
+  "azureLoginModal.azureReadMore": "Learn more about the Azure free account. Read the FAQ >",
+  "azureLoginModal.createAccount": "Create Free Account",
+  "azureLoginModal.freeAccountAnswer": "All you need is a phone number, a credit or debit card, and a GitHub account or Microsoft account username (formerly Windows Live ID).",
+  "azureLoginModal.freeAccountQuestion": "What do I need to sign up for a free account?",
+  "azureLoginModal.freeTrialAnswer": "TODO",
```
Comment (xtinah-w) [inline src/client/src/translations/en.json]:
```suggestion
  "azureLoginModal.freeTrialAnswer": "After 12 months, any of these products you may be using will continue to run, and you’ll be billed at the standard pay-as-you-go rates.",
```

File: src/client/src/translations/en.json [hunk 5]
```diff
@@ -39,6 +39,18 @@
...
+  "azureLoginModal.freeTrialQuestion": "What happens if I want to upgrade my free trial?",
+  "azureLoginModal.freeTrialUpgradeAnswer": "We’ll notify you so you can decide if you want to upgrade to pay-as-you-go pricing and remove the spending limit. If you do, you’ll have access to all the free products. If you don’t, your account and products will be disabled, and you'll need to upgrade to resume usage.",
+  "azureLoginModal.freeTrialUpgradeQuestion": "Do I get charged once I finish my free trail?",
```
Comment (xtinah-w) [inline src/client/src/translations/en.json]:
```suggestion
  "azureLoginModal.freeTrialUpgradeQuestion": "Do I get charged once I finish my free trial?",
```

File: src/client/src/containers/AzureLoginModal/strings.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    freeTrialUpgradeQuestion: {
+        defaultMessage: "What happens if I want to upgrade my free trial?",
+        id: "azureLoginModal.freeTrialQuestion"
+    },
+    freeTrialAnswer: {
+        defaultMessage: "We’ll notify you so you can decide if you want to upgrade to pay-as-you-go pricing and remove the spending limit. If you do, you’ll have access to all the free products. If you don’t, your account and products will be disabled, and you'll need to upgrade to resume usage.",
+        id: "azureLoginModal.freeTrialUpgradeAnswer"
+    },
+    freeTrialQuestion: {
+        defaultMessage: "Do I get charged once I finish my free trail?",
```
Comment (xtinah-w) [inline src/client/src/containers/AzureLoginModal/strings.ts]:
```suggestion
        defaultMessage: "Do I get charged once I finish my free trial?",
```

File: src/client/src/translations/en.json (hunk 5 above)
Comment (Tanya0609) [reply src/client/src/translations/en.json]: Thanks 👍

File: src/client/src/containers/AzureLoginModal/strings.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        defaultMessage: "All you need is a phone number, a credit or debit card, and a GitHub account or Microsoft account username (formerly Windows Live ID).",
+        id: "azureLoginModal.freeAccountAnswer"
+    },
+    freeAccountQuestion: {
+        defaultMessage: "What do I need to sign up for a free account?",
+        id: "azureLoginModal.freeAccountQuestion"
+    },
+    freeTrialUpgradeAnswer: {
+        defaultMessage: "TODO",
+        id: "azureLoginModal.freeTrialAnswer"
```
Comment (jasmingoh) [inline src/client/src/containers/AzureLoginModal/strings.ts]: should be azureLoginModal.freeTrialUpgradeAnswer

File: src/client/src/containers/AzureLoginModal/strings.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    },
+    freeTrialUpgradeQuestion: {
+        defaultMessage: "What happens if I want to upgrade my free trial?",
+        id: "azureLoginModal.freeTrialQuestion"
```
Comment (jasmingoh) [inline src/client/src/containers/AzureLoginModal/strings.ts]: should be azureLoginModal.freeTrialUpgradeQuestion

File: src/client/src/containers/AzureLoginModal/strings.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    },
+    freeTrialAnswer: {
+        defaultMessage: "We’ll notify you so you can decide if you want to upgrade to pay-as-you-go pricing and remove the spending limit. If you do, you’ll have access to all the free products. If you don’t, your account and products will be disabled, and you'll need to upgrade to resume usage.",
+        id: "azureLoginModal.freeTrialUpgradeAnswer"
```
Comment (jasmingoh) [inline src/client/src/containers/AzureLoginModal/strings.ts]: should be azureLoginModal.freeTrialAnswer

File: src/client/src/containers/AzureLoginModal/strings.ts [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    },
+    freeTrialQuestion: {
+        defaultMessage: "Do I get charged once I finish my free trial?",
+        id: "azureLoginModal.freeTrialUpgradeQuestion"
```
Comment (jasmingoh) [inline src/client/src/containers/AzureLoginModal/strings.ts]: should be azureLoginModal.freeTrialQuestion

File: src/client/src/containers/AzureLoginModal/strings.ts [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    freeAccountAnswer: {
+        defaultMessage: "All you need is a phone number, a credit or debit card, and a GitHub account or Microsoft account username (formerly Windows Live ID).",
+        id: "azureLoginModal.freeAccountAnswer"
+    },
+    freeAccountQuestion: {
+        defaultMessage: "What do I need to sign up for a free account?",
+        id: "azureLoginModal.freeAccountQuestion"
+    },
+    freeTrialUpgradeAnswer: {
+        defaultMessage: "TODO",
```
Comment (jasmingoh) [inline src/client/src/containers/AzureLoginModal/strings.ts]: @xtinah-w please help!

File: src/client/src/components/CollapsibleInfoBox/index.tsx [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+  };
+
+  return (
+    <div>
+      <div className={styles.questionTitle}>
+        {question}
+        {isAnswerShown ? (
+          <Up className={styles.toggleIcon} onClick={toggleAnswerShown} onKeyDown={keyDownHandler} />
+        ) : (
+            <Down className={styles.toggleIcon} onClick={toggleAnswerShown} onKeyDown={keyDownHandler} />
```
Comment (jsondoo) [inline src/client/src/components/CollapsibleInfoBox/index.tsx]: if you put `tabIndex={0}` here, this should work

File: src/client/src/containers/AzureLoginModal/index.tsx (hunk 2 above)
Comment (Tanya0609) [reply src/client/src/containers/AzureLoginModal/index.tsx:77]: I added the tab in my PR :) now it is togglable via tab + enter

File: src/client/src/components/CollapsibleInfoBox/index.tsx (hunk 12 above)
Comment (Tanya0609) [reply src/client/src/components/CollapsibleInfoBox/index.tsx]: I was doing this on Mac, it failed the build. But now I can do it on my Windows, thanks! PR updated.

File: src/client/src/containers/AzureLoginModal/strings.ts (hunk 11 above)
Comment (Tanya0609) [reply src/client/src/containers/AzureLoginModal/strings.ts]: the suggestions did not add automatically, I just manually added. PR updated :)

File: src/client/src/containers/AzureLoginModal/strings.ts (hunk 10 above)
Comment (Tanya0609) [reply src/client/src/containers/AzureLoginModal/strings.ts]: PR updated :)

File: src/client/src/containers/AzureLoginModal/strings.ts (hunk 9 above)
Comment (Tanya0609) [reply src/client/src/containers/AzureLoginModal/strings.ts]: Great catch, PR updated :)

File: src/client/src/containers/AzureLoginModal/strings.ts (hunk 8 above)
Comment (Tanya0609) [reply src/client/src/containers/AzureLoginModal/strings.ts]: PR updated :)

File: src/client/src/containers/AzureLoginModal/strings.ts (hunk 7 above)
Comment (Tanya0609) [reply src/client/src/containers/AzureLoginModal/strings.ts]: PR updated :)

File: src/client/src/components/CollapsibleInfoBox/index.tsx [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+          />
+        )}
+      </div>
+
+      {isAnswerShown && <div className={styles.question}>{answer}</div>}
+    </div>
+  );
+};
+
+export default injectIntl(CollapsibleInfoBox);
```
Comment (streetcoder123) [inline src/client/src/components/CollapsibleInfoBox/index.tsx]: Why do you wrap the component in `injectIntl` when you don't need it anywhere in the component?

File: src/client/src/components/CollapsibleInfoBox/index.tsx [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+import styles from "./styles.module.css";
+import { KEY_EVENTS } from "../../utils/constants";
+
+interface IProps {
+  question: string;
+  answer: string;
+  initialAnswerShownState?: boolean;
+}
+
+type Props = IProps & InjectedIntlProps;
```
Comment (streetcoder123) [inline src/client/src/components/CollapsibleInfoBox/index.tsx]: Which prop is of type `InjectedIntlProps`? If there is none, we can probably remove this.

Comment (Tanya0609) [reply src/client/src/components/CollapsibleInfoBox/index.tsx]: Great catch, removed :)

File: src/client/src/components/CollapsibleInfoBox/index.tsx (hunk 13 above)
Comment (Tanya0609) [reply src/client/src/components/CollapsibleInfoBox/index.tsx]: Great catch, removed :)

**Review by streetcoder123 — APPROVED** [review]

**Review by jasmingoh — APPROVED** [review]

**Review by SahilTara — APPROVED** [review]

**Review by piotrmark — APPROVED** [review]

**Pull request merged**